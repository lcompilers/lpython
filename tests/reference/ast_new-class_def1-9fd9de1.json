{
    "basename": "ast_new-class_def1-9fd9de1",
    "cmd": "lpython --show-ast --indent --new-parser --no-color {infile} -o {outfile}",
    "infile": "tests/parser/class_def1.py",
    "infile_hash": "e904e820fab9d596d2675abf65e19fc2b5fb6ecf950a902a0ec537fb",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast_new-class_def1-9fd9de1.stdout",
    "stdout_hash": "9802c1d135072a26d8cc97d779d5c89b93317138aa7903dde7d42fb0",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}