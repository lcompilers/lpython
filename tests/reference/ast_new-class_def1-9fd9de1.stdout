(Module
    [(ClassDef
        Person
        []
        []
        [(FunctionDef
            __init__
            ([]
            [(self
            ()
            ())
            (name
            ()
            ())
            (age
            ()
            ())]
            []
            []
            []
            []
            [])
            [(Assign
                [(Attribute
                    (Name
                        self
                        Load
                    )
                    name
                    Store
                )]
                (Name
                    name
                    Load
                )
                ()
            )
            (Assign
                [(Attribute
                    (Name
                        self
                        Load
                    )
                    age
                    Store
                )]
                (Name
                    age
                    Load
                )
                ()
            )]
            []
            ()
            ()
        )
        (FunctionDef
            print_name
            ([]
            [(self
            ()
            ())]
            []
            []
            []
            []
            [])
            [(Pass)
            (Expr
                (Call
                    (Name
                        print
                        Load
                    )
                    [(BinOp
                        (ConstantStr
                            "Hello my name is "
                            ()
                        )
                        Add
                        (Attribute
                            (Name
                                self
                                Load
                            )
                            name
                            Load
                        )
                    )]
                    []
                )
            )]
            []
            ()
            ()
        )]
        []
    )
    (ClassDef
        Dog
        []
        []
        [(FunctionDef
            __init__
            ([]
            [(self
            ()
            ())
            (name
            ()
            ())]
            []
            []
            []
            []
            [])
            [(Assign
                [(Attribute
                    (Name
                        self
                        Load
                    )
                    name
                    Store
                )]
                (Name
                    name
                    Load
                )
                ()
            )
            (Assign
                [(Attribute
                    (Name
                        self
                        Load
                    )
                    tricks
                    Store
                )]
                (List
                    []
                    Load
                )
                ()
            )]
            []
            ()
            ()
        )
        (FunctionDef
            add_trick
            ([]
            [(self
            ()
            ())
            (trick
            ()
            ())]
            []
            []
            []
            []
            [])
            [(Expr
                (Call
                    (Attribute
                        (Attribute
                            (Name
                                self
                                Load
                            )
                            tricks
                            Load
                        )
                        append
                        Load
                    )
                    [(Name
                        trick
                        Load
                    )]
                    []
                )
            )]
            []
            ()
            ()
        )]
        []
    )
    (ClassDef
        Mapping
        []
        []
        [(FunctionDef
            __init__
            ([]
            [(self
            ()
            ())
            (iterable
            ()
            ())]
            []
            []
            []
            []
            [])
            [(Assign
                [(Attribute
                    (Name
                        self
                        Load
                    )
                    items_list
                    Store
                )]
                (List
                    []
                    Load
                )
                ()
            )
            (Expr
                (Call
                    (Attribute
                        (Name
                            self
                            Load
                        )
                        __update
                        Load
                    )
                    [(Name
                        iterable
                        Load
                    )]
                    []
                )
            )]
            []
            ()
            ()
        )
        (FunctionDef
            update
            ([]
            [(self
            ()
            ())
            (iterable
            ()
            ())]
            []
            []
            []
            []
            [])
            [(For
                (Name
                    item
                    Store
                )
                (Name
                    iterable
                    Load
                )
                [(Expr
                    (Call
                        (Attribute
                            (Attribute
                                (Name
                                    self
                                    Load
                                )
                                items_list
                                Load
                            )
                            append
                            Load
                        )
                        [(Name
                            item
                            Load
                        )]
                        []
                    )
                )]
                []
                ()
            )]
            []
            ()
            ()
        )
        (Assign
            [(Name
                __update
                Store
            )]
            (Name
                update
                Load
            )
            ()
        )]
        []
    )
    (ClassDef
        MappingSubclass
        [(Name
            Mapping
            Load
        )]
        []
        [(FunctionDef
            update
            ([]
            [(self
            ()
            ())
            (keys
            ()
            ())
            (values
            ()
            ())]
            []
            []
            []
            []
            [])
            [(For
                (Name
                    item
                    Store
                )
                (Call
                    (Name
                        zip
                        Load
                    )
                    [(Name
                        keys
                        Load
                    )
                    (Name
                        values
                        Load
                    )]
                    []
                )
                [(Expr
                    (Call
                        (Attribute
                            (Attribute
                                (Name
                                    self
                                    Load
                                )
                                items_list
                                Load
                            )
                            append
                            Load
                        )
                        [(Name
                            item
                            Load
                        )]
                        []
                    )
                )]
                []
                ()
            )]
            []
            ()
            ()
        )]
        []
    )
    (FunctionDef
        check
        ([]
        []
        []
        []
        []
        []
        [])
        [(Assign
            [(Name
                p
                Store
            )]
            (Call
                (Name
                    Person
                    Load
                )
                [(ConstantStr
                    "John"
                    ()
                )
                (ConstantInt
                    36
                    ()
                )]
                []
            )
            ()
        )
        (Assign
            [(Name
                d
                Store
            )]
            (Call
                (Name
                    Dog
                    Load
                )
                [(ConstantStr
                    "Fido"
                    ()
                )]
                []
            )
            ()
        )
        (Expr
            (Call
                (Attribute
                    (Name
                        d
                        Load
                    )
                    add_trick
                    Load
                )
                [(ConstantStr
                    "roll over"
                    ()
                )]
                []
            )
        )
        (Assign
            [(Name
                m
                Store
            )]
            (Call
                (Name
                    Mapping
                    Load
                )
                [(ConstantStr
                    "x"
                    ()
                )]
                []
            )
            ()
        )
        (Expr
            (Call
                (Attribute
                    (Name
                        m
                        Load
                    )
                    update
                    Load
                )
                [(ConstantStr
                    "y"
                    ()
                )]
                []
            )
        )]
        []
        ()
        ()
    )
    (Expr
        (Call
            (Name
                check
                Load
            )
            []
            []
        )
    )]
    []
)
